test/occupation.jl

```diff
@@ -46,7 +48,7 @@ end
     # See that the electron count still works if we add temperature
     Ts = (0, 1e-6, .1, 1.0)
     for T in Ts, meth in DFTK.Smearing.smearing_methods
-        model = Model(silicon.lattice; n_electrons=silicon.n_electrons, temperature=T, smearing=meth())
+        model = Model(silicon.lattice; n_electrons=silicon.n_electrons, temperature=T * 1u"Eh_au", smearing=meth())
```

Comment:
That's exactly the sort of type annotations we wanted to avoid.

From my point of view units are a convenience feature and strictly "opt-in". The idea is that we use atomic units internally in the code and so if nothing is specified it should just default to being atomic.